libsolidity/ast/AST.h

```diff
@@ -66,13 +66,15 @@ class ASTNode: private boost::noncopyable
 public:
 	using SourceLocation = langutil::SourceLocation;
 
-	explicit ASTNode(SourceLocation const& _location);
+	explicit ASTNode(int64_t _id, SourceLocation const& _location);
 	virtual ~ASTNode();
 
 	/// @returns an identifier of this AST node that is unique for a single compilation run.
-	size_t id() const { return m_id; }
+	int64_t id() const { return m_id; }
 	/// Resets the global ID counter. This invalidates all previous IDs.
 	static void resetID();
+	/// Returns the next ID, increasing the global counte
```

Comment:
```suggestion
	/// Returns the next ID, increasing the global counter
```

Reply:
Actually ``resetID`` and ``nextID``  should probably be moved to the parser instead, don't yout think?

Reply:
Well, the `CompilerStack.cpp` also needs access to the `resetID()` part. Wasn't sure what the best place was

Reply:
where as the parser doesn't actually need `resetID()`

Reply:
What would prevent the CompilerStack from just calling ``Parser::resetID()`` instead of ``ASTNode::resetID()``?

Reply:
Actually if you make the whole thing local to the Parser, you don't need any singleton and any resetting at all, since ``CompilerStack::parse`` (which does the reset) always creates a new parser object anyways, right? So the whole IDDispenser can go away and move into the parser...